package/butterfly/lib/dev_server.dart

```diff
@@ -123,29 +120,36 @@ class ButterflyDevServer {
     request.response.write(JSON.encode(result));
   }
 
-  Future<Null> _serveDevServerClientFile(HttpRequest request) async {
+  /// Mimics pub serve's behavior of service package-relative static files.
+  Future<Null> _servePackageAsset(HttpRequest request) async {
     final packagesFile = new File('.packages');
 
     if (!await packagesFile.exists()) {
       throw new StateError('.packages file not found. Have you run pub get?');
     }
 
+    final pathFragments = pathlib.split(request.uri.path);
+    assert(pathFragments[0] == '/');
+    assert(pathFragments[1] == 'packages');
+    final packageName = pathFragments[2];
+    final pathWithinPackage = pathFragments.skip(3).join(pathlib.separator);
+
     List<String> packageInfoParts = await packagesFile
         .openRead()
         .transform(const Utf8Decoder())
         .transform(const LineSplitter())
```

Comment:
I think it might be better to fuse these two codecs together, even though they can't be const